torchio/transforms/preprocessing/spatial/cropOrPad.py

```diff
@@ -0,0 +1,151 @@
+            break
+        return data
+
+    @staticmethod
+    def _get_six_bounds_parameters(parameters: np.ndarray):
+        r"""Compute bounds parameters for ITK filters.
+
+        Args:
+            parameters: Tuple :math:`(d, h, w)` with the number of voxels to be
+                cropped or padded.
+
+        Returns:
+            Tuple :math:`(d_{ini}, d_{fin}, h_{ini}, h_{fin}, w_{ini}, w_{fin})`,
+            where :math:`n_{ini} = \left \lceil \frac{n}{2} \right \rceil` and
+            :math:`n_{fin} = \left \lfloor \frac{n}{2} \right \rfloor`.
+
+        Example:
+            >>> p = np.array((4, 0, 7))
+            >>> _get_six_bounds_parameters(p)
+            (2, 2, 0, 0, 4, 3)
+
+        """
+        parameters = parameters / 2
+        result = []
+        for n in parameters:
+            ini, fin = int(np.ceil(n)), int(np.floor(n))
+            result.extend([ini, fin])
+        return tuple(result)
+
+    def _compute_center_crop_or_pad(self, sample: dict):
+        source_shape = self._get_sample_shape(sample)
+        # The parent class turns the 3-element shape tuple (d, h, w)
+        # into a 6-element bounds tuple (d, d, h, h, w, w)
+        target_shape = np.array(self.bounds_parameters[::2])
+        diff_shape = target_shape - source_shape
+
+        cropping = -np.minimum(diff_shape, 0)
+        if cropping.any():
+            cropping_params = self._get_six_bounds_parameters(cropping)
+
+        padding = np.maximum(diff_shape, 0)
+        if padding.any():
+            padding_params = self._get_six_bounds_parameters(padding)
+
+        return padding_params, cropping_params
+
+    def _compute_mask_center_crop_or_pad(self, sample: dict):
+        mask = sample[self.mask_key][DATA].numpy()
+        # Original sample shape (from mask shape)
+        sample_shape = np.squeeze(mask).shape
+        # Calculate the bounding box of the brain
+        xmin, xmax, ymin, ymax, zmin, zmax = self._bbox_mask(np.squeeze(mask))
+        # Coordinates of the center of the brain
+        center_mask = ((xmax - xmin) / 2 + xmin, (ymax - ymin) / 2 + ymin, (zmax - zmin) / 2 + zmin)
+        # List of padding to do
+        padding = []
+        # Final cropping (after padding)
+        cropping = []
+        for dim, center_dim in enumerate(center_mask):
+            # Compute the coordinates of the target shape taken from the center of the mask
+            begin = center_dim - (self.bounds_parameters[2 * dim] / 2)
+            end = center_dim + (self.bounds_parameters[2 * dim + 1] / 2)
+            # Check if dimension needs padding (before or after)
+            begin_pad = abs(np.minimum(begin, 0))
+            end_pad = np.maximum(end - sample_shape[dim], 0)
+            # Check if cropping needed
+            begin_crop = abs(np.round(abs(np.maximum(begin, 0)))).astype(np.uint)
+            end_crop = abs(np.round(np.minimum(end - sample_shape[dim], 0))).astype(np.uint)
```

Comment:
Can you replace all `np.round` by `round`? That way, you won't need to convert to `int`. Also, I think you can use `min` instead of `np.minimum`.

Reply:
And `max` instead of `np.maximum`.

Reply:
Actually, the returned values from all these operations (without `np`) are floats. I will replace these operations but it will still require a final casting operation

Reply:
Did you end up removing the casting?

Reply:
I did not remove it, I moved further in the function (it does not work otherwise)